src/arglist.c

```diff
@@ -683,6 +683,12 @@ do_argfile(exarg_T *eap, int argn)
     char_u	*p;
     int		old_arg_idx = curwin->w_arg_idx;
 
+    if (!eap->forceit && curwin->w_p_stb)
```

Comment:
> if (!eap->forceit && curwin->w_p_stb)

This part with the error message should be in a seperat function so you do not duplicate

Reply:
To make sure I understand your suggestion, do you want a common function that contains both if-logic + an error message that prints to Vim on-fail and change the code like ...

```c
int is_allowed_to_go_to_buffer(exarg_T *eap)
{
    if (!eap->forceit && curwin->w_p_stb)
    {
        semsg(_("E922: Cannot go to buffer. 'switchbuf' is enabled. Use ! to force it."))
        return FALSE;
    }

    return TRUE;
}
```

Then in callers ...

```c
if (!is_allowed_to_go_to_buffer(eap)
    return;
```

Is this along the lines of what you wanted or did you want something different? If we change the signature to `int is_allowed_to_go_to_buffer(int forceit)` we could also probably use the same function in other `.c` files too

Related to this, I was also wondering if you'd like any of the currently hard-coded messages to be put into `src/errors.h`

Reply:
yes to both please :)

Reply:
This is now done